Source/Core/DolphinQt/DiscordJoinRequestDialog.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#ifdef USE_DISCORD_PRESENCE
+
+#include "DiscordJoinRequestDialog.h"
+
+#include <QGridLayout>
+#include <QLabel>
+#include <QPixmap>
+#include <QPushButton>
+#include <discord-rpc/include/discord_rpc.h>
+#include "Common/HttpRequest.h"
+
+DiscordJoinRequestDialog::DiscordJoinRequestDialog(QWidget* parent, const char id[],
+                                                   const std::string& discord_tag,
+
+                                                   const char avatar[])
+    : QDialog(parent), m_user_id(id), m_close_timestamp(std::time(nullptr) + m_max_lifetime)
+{
+  setWindowTitle(tr("Request to Join Your Party"));
+  setWindowFlags(windowFlags() & ~Qt::WindowContextHelpButtonHint);
+
+  QPixmap avatar_pixmap;
+
+  if (avatar[0] != '\0')
+  {
+    const std::string path = "https://cdn.discordapp.com/avatars/";
```

Comment:
If you include `"Common/StringUtil.h"`, this can likely just be compressed to:

```cpp
const std::string avatar_endpoint = StringFromFormat("https://cdn.discordapp.com/avatars/%s/%s.png", id, avatar);
```